compose/ui/ui/src/uikitInstrumentedTest/kotlin/android/compose/ui/test/UIKitInstrumentedTest.kt

```diff
@@ -162,39 +162,45 @@ internal class UIKitInstrumentedTest {
         return !hadSnapshotChanges && !hasPendingRecomposer && !isApplyObserverNotificationPending
     }
 
-    @OptIn(ExperimentalNativeApi::class)
-    fun waitForIdle(deadline: Duration = 5.seconds): Boolean {
+    fun waitForIdle(timeoutMillis: Long = 5_000) {
         val defaultIdleScore = 10
         var idleScore = defaultIdleScore
-        val timeoutAchieved = waitForExpectation(deadline = deadline) {
+        waitUntil(
+            conditionDescription = "waitForIdle: timeout ${timeoutMillis}ms reached.",
+            timeoutMillis = timeoutMillis
+        ) {
             if (isIdle) {
                 idleScore -= 1
             } else {
                 idleScore = defaultIdleScore
             }
             idleScore == 0
         }
-        assert(timeoutAchieved) { "waitForIdle reached $deadline deadline" }
-        return timeoutAchieved
     }
 
-    fun waitForExpectation(deadline: Duration = 5.seconds, expectation: () -> Boolean): Boolean {
+    @OptIn(ExperimentalNativeApi::class)
+    fun waitUntil(
+        conditionDescription: String? = null,
+        timeoutMillis: Long = 5_000,
+        condition: () -> Boolean
+    ) {
         val runLoop = NSRunLoop.currentRunLoop()
-        var remainingTime = deadline
+        var remainingTime = timeoutMillis.milliseconds
         var time = NSDate.timeIntervalSinceReferenceDate()
-        while (!expectation() && remainingTime > 0.seconds) {
```

Comment:
No need for two variables. You can just save
```
val startTime = NSDate.timeIntervalSinceReferenceDate()
```
and check
```
NSDate.timeIntervalSinceReferenceDate() - startTime < timeout
```

Also, why not use `Clock.System`, or even better, `TimeSource.Monotonic`?

Reply:
Indeed, we can remove one variable, but the condition check must be different, taking into account time between `while` and `assert` checks.
TimeSource. Monotonic - it looks like it can be good replacement for standard API.